lib/reek/code_comment.rb

```diff
@@ -24,6 +46,13 @@ def initialize(comment)
       @config            = Hash.new { |hash, key| hash[key] = {} }
 
       @original_comment.scan(CONFIGURATION_REGEX) do |smell_type, options|
+        # We need the legacy separator in front of the options string for
+        # the "Yaml.load" below to work properly. With the ":" in front of the options
+        # "Yaml.load" will return something like this:
+        #   {"TooManyMethods"=>{"max_methods"=>30}}
+        # Without it something bogus like this:
+        #  {"TooManyMethods { max_methods"=>"30 }"}
+        options.prepend(LEGACY_SEPARATOR) if options && !options.start_with?(LEGACY_SEPARATOR)
```

Comment:
... and we could even do something like this (untested), and avoid the `LEGACY_SEPARATOR` complications:

```ruby
@config.merge! smell_type => YAML.load(options || DISABLE_DETECTOR_CONFIGURATION)
```